src/components/settings/data/hooks.js

```diff
@@ -60,7 +61,59 @@ export const useLinkManagement = (enterpriseUUID) => {
   };
 };
 
+const initialCustomerAgreementState = {
+  netDaysUntilExpiration: 0,
+};
+/*
+ * @param {Objet {enterpriseId: string}}
+ * @returns {customerAgreement: Object, loadingCustomerAgreement: bool}
+ * customerAgreement:{
+ *  netDaysUntilExpiration: number
+ * }
+ */
+export const useCustomerAgreementData = ({ enterpriseId }) => {
+  const [customerAgreement, setCustomerAgreement] = useState(initialCustomerAgreementState);
+  const [loadingCustomerAgreement, setLoadingCustomerAgreement] = useState(true);
+
+  const loadCustomerAgreementData = (page = 1) => {
+    if (!loadingCustomerAgreement) {
+      setLoadingCustomerAgreement(true);
+    }
```

Comment:
nit: I think this could probably be simplified to just `setLoadingCustomerAgreement(true)`; not sure the `if` is really adding much here?